### DISCUSSION TITLE
Notification Service Features

### DISCUSSION THREAD

**Discussion by gysenthil** [Sunbird-Lern/Community#101] [Query] [answered]
We are in the need of notification service to send notifications on specific events like course creation, course completion etc..
I am using the branch - release-5.0.0
I have tried some of the APIS  like
POST /v1/notification/send				   @controllers.notification.NotificationController.sendNotification()
POST /v1/notification/send/sync			   @controllers.notification.NotificationController.sendSyncNotification()
I am getting the Bad Request Error. I have provided the request with the help of documentation.
Need help in getting the proper api request for sending Notifications.
Also Need inputs on integrating with other services for sending the notifications.

**Comment by reshmi-nair** [comment]
@gysenthil Will check and share the details soon

**Reply by reshmi-nair** [reply, marked as answer]
@gysenthil Please try with below request:
{
    "request": {
        "mode": "test",
        "body": "if mode SMS then complete body will go as sms to that recipientUserIds",
        "emailTemplateType": "defaultCertTemp",
        "subject": "Course completion certificate",
        "stateImgUrl": "https://test.com/img/File-0128212938260643843.png",
        "firstName": "name of user<String>",
        "TraningName": "name of attended traning",
        "heldDate": "traning held date (25-08-2019)",
        "pdfUrl": "https://test.com/img/File-0128231581485547521.pdf",
        "regardsperson": "Chairperson",
        "regards": "Director", 
        "recipientEmails": [
            "test@test.com"
        ]         
    }
}

To integrate in other services, if it is a sync call, then you can directly call this API from that service. If it is async, then setup notification flink job from data-pipeline repo. [data-pipeline](https://github.com/Sunbird-Lern/data-pipeline/tree/release-5.2.0/notification). Post the events to kafka topic "lms.notification1" It will send async notification
